_source/_posts/2022-02-25-thymeleaf-security.md

```diff
@@ -0,0 +1,670 @@
+
+Among its features, Thymeleaf allows to:
+
+- Work with fragments: only a part of a template can be rendered, which can be useful for updating part of a page from responses in AJAX calls. It also provides a componentization tool, as the fragments can be included in multiple templates.
+- Handle forms with model objects containing their fields
+- Render variables and externalized text messages through its [Standard Expression Syntax](https://www.thymeleaf.org/doc/tutorials/3.0/usingthymeleaf.html#standard-expression-syntax)
+- Perform iterations and conditional evaluations
+
+{% img blog/thymeleaf-security/thymeleaf-logo-white.png alt:"Thymeleaf Logo" width:"600" %}{: .center-image }
+
+# Create a Spring Boot Webflux application with Thymeleaf
+
+Create a simple monolithic Spring Boot reactive application, that will use Thymeleaf for page templates.
+You can use [Spring Initializr](https://start.spring.io/), from its web UI, or download the starter app with the following HTTPie line:
+
+```shell
+https -d start.spring.io/starter.zip bootVersion==2.6.4 \
+  baseDir==thymeleaf-security \
+  groupId==com.okta.developer.thymeleaf-security \
+  artifactId==thymeleaf-security \
+  name==thymeleaf-security \
+  packageName==com.okta.developer.demo \
+  javaVersion==11 \
+  dependencies==webflux,okta,thymeleaf
+```
+
+Extract the Maven project, and some additional dependencies. `thymeleaf-extras-springsecurity5` dependency is required to include security conditionals in the templates, and
+`spring-security-test` has mock login features that will help when writing controller tests.
+
+```xml
+<dependency>
+  <groupId>org.thymeleaf.extras</groupId>
+  <artifactId>thymeleaf-extras-springsecurity5</artifactId>
+  <version>3.0.4.RELEASE</version>
+</dependency>
+<dependency>
+  <groupId>org.springframework.security</groupId>
+  <artifactId>spring-security-test</artifactId>
+  <scope>test</scope>
+</dependency>
+```
+
+## Add Okta authentication
+
+{% include setup/cli.md type="web" framework="Okta Spring Boot Starter" %}
+
+
+
+The Okta configuration is now in `application.properties`. Rename it to `application.yml` and add the following additional properties:
+
+```yml
+spring:
+  security:
+    oauth2:
+      client:
+        provider:
+          okta:
+            user-name-attribute: email
+
+okta:
+  oauth2:
+    issuer: https://{yourOktaDomain}/oauth2/default
+    client-id: {clientId}
+    client-secret: {clientSecret}
+    scopes:
+      - email
+      - openid
```

Comment:
Are you sure this works? We use `scope: openid,profile,email` in JHipster. https://github.com/jhipster/generator-jhipster/blob/main/generators/server/templates/src/main/resources/config/application.yml.ejs#L302

Reply:
Only `openid` is a required scope for an OpenID Connect request.
https://openid.net/specs/openid-connect-basic-1_0.html#Scopes
It worked for me. Will add a comment in the post.
From the docs: "`profile` OPTIONAL. This scope value requests access to the End-User's default profile Claims, which are: name, family_name, given_name, middle_name, nickname, preferred_username, profile, picture, website, gender, birthdate, zoneinfo, locale, and updated_at."